ESSArch_Core/WorkflowEngine/tests/test_tasks.py

```diff
@@ -236,3 +239,182 @@ def test_retry_with_args(self):
+        }
+        mock_logger_log.assert_called_once_with(logging.ERROR, mock.ANY, extra=expected_extra)
+
+    @mock.patch("ESSArch_Core.WorkflowEngine.models.ProcessTask.objects.filter")
+    @mock.patch("ESSArch_Core.WorkflowEngine.dbtask.DBTask.create_event")
+    def test_success_when_event_type_not_none_then_create_event(self, mock_create_event, mock_process_task):
+        db_task = DBTask()
+        db_task.event_type = not None
+        task_id = uuid.uuid4()
+        retval = uuid.uuid4()
+        args = uuid.uuid4()
+        kwargs = uuid.uuid4()
+
+        db_task.success(retval, task_id, args, kwargs)
+
+        mock_process_task.assert_called_once()
+        mock_create_event.assert_called_once_with(task_id, celery_states.SUCCESS, args, kwargs, None, retval)
+
+    @mock.patch("ESSArch_Core.WorkflowEngine.models.ProcessTask.objects.filter")
+    @mock.patch("ESSArch_Core.WorkflowEngine.dbtask.DBTask.create_event")
+    def test_success_when_event_type_is_none_dont_create_event(self, mock_create_event, mock_process_task):
+        db_task = DBTask()
+        db_task.event_type = None
+        task_id = uuid.uuid4()
+        retval = uuid.uuid4()
+        args = uuid.uuid4()
+        kwargs = uuid.uuid4()
+
+        db_task.success(retval, task_id, args, kwargs)
+
+        mock_create_event.assert_not_called()
+        mock_process_task.assert_called_once()
+
+    @mock.patch("ESSArch_Core.WorkflowEngine.models.ProcessTask.objects.filter")
+    @mock.patch("ESSArch_Core.WorkflowEngine.dbtask.DBTask.create_event")
+    def test_success_when_track_is_False_then_return(self, mock_create_event, mock_process_task):
+        db_task = DBTask()
+        db_task.track = False
+        task_id = uuid.uuid4()
+        retval = uuid.uuid4()
+        args = uuid.uuid4()
+        kwargs = uuid.uuid4()
+
+        db_task.success(retval, task_id, args, kwargs)
+
+        mock_create_event.assert_not_called()
+        mock_process_task.assert_not_called()
+
+    @mock.patch("ESSArch_Core.WorkflowEngine.models.ProcessTask.objects.filter")
+    @mock.patch("ESSArch_Core.WorkflowEngine.dbtask.DBTask.create_event")
+    def test_success_when_chunk_is_True_then_return(self, mock_create_event, mock_process_task):
+        db_task = DBTask()
+        db_task.chunk = True
+        task_id = uuid.uuid4()
+        retval = uuid.uuid4()
+        args = uuid.uuid4()
+        kwargs = uuid.uuid4()
+
+        db_task.success(retval, task_id, args, kwargs)
+
+        mock_create_event.assert_not_called()
+        mock_process_task.assert_not_called()
+
+    @mock.patch("billiard.einfo.ExceptionInfo")
+    @mock.patch("ESSArch_Core.WorkflowEngine.models.ProcessTask.objects.filter")
+    @mock.patch("ESSArch_Core.WorkflowEngine.dbtask.DBTask.create_event")
+    def test_failure_when_event_type_not_none_then_create_event(self, mock_create_event, mock_proc_task, mock_einfo):
+        db_task = DBTask()
+        db_task.event_type = not None
```

Comment:
`event_type` is either `None` or an `int`

```suggestion
        db_task.event_type = 123
```